lib/galaxy_test/api/test_datasets.py

```diff
@@ -134,3 +135,30 @@ def test_tag_change(self):
         assert 'tag_c' in updated_hda['tags']
         assert 'name:tag_d' in updated_hda['tags']
         assert 'name:tag_e' in updated_hda['tags']
+
+    def test_update_datatype(self):
+        hda_id = self.dataset_populator.new_dataset(self.history_id)['id']
+        original_hda = self._get(
+            "histories/{history_id}/contents/{hda_id}".format(history_id=self.history_id, hda_id=hda_id)).json()
+        assert original_hda['extension'] == 'txt'
+        assert original_hda['data_type'] == 'galaxy.datatypes.data.Text'
+        assert len(original_hda['visualizations']) < 10
```

Comment:
Maybe checking that a tabular-specific visualization is not present in `original_hda['visualizations']` is more future-proof? Same (reversed) below.